File: hw/fuentes/U3/HW20COMPOSITE/HW20COMPOSITE_PY/ec/edu/espe/enterprise/model/Supervisor.py
from ec.edu.espe.enterprise.model.Employee import Employee

class Supervisor(Employee):
    def __init__(self, name="not assigned yet", title="not assigned yet"):
        super().__init__(name, title)
        self.directReports = []

    def add(self, anEmployee):
        self.directReports.append(anEmployee)

    def state_name(self):
        super().state_name()
        for emp in self.directReports:
            emp.state_name()

File: hw/chicaiza/u3/HW20Composite/CompositePatternJS/ec/edu/espe/model/Supervisor.js
import { Employee } from './Employee.js';

export class Supervisor extends Employee {
  constructor(name = "", title = "") {
    super(name, title);
    this.directReports = [];
  }

  stateName() {
    super.stateName();
    this.directReports.forEach(employee => {
      employee.stateName();
    });
  }

  add(employee) {
    this.directReports.push(employee);
  }
}
